rfc/2023/2023-05-01-va-rfc-lightweight-apps-paths.md

```diff
@@ -0,0 +1,96 @@
+
+The VA.gov Platform provides [few options for creating new frontend (FE) applications and features](https://depo-platform-documentation.scrollhelp.site/developer-docs/adding-applications-and-features). As the Frontend Support Dashboard shows, most applications are loading multiple large JavaScript bundles that impact performance by increasing page load times, even when the application does not use the code those bundles provide. The VA.gov Platform needs to support options for smaller frontend applications.
+
+
+## Motivation
+
+>Which problem(s) does this solve? Why? Use active voice, present tense, and decisive language.
+> Example: Framework XYZ is a modern, well-known, web rendering framework. It is easy to learn, easy to use, and provides high-performance web rendering. etc
+
+Static-site generation techniques allow us to build smaller, faster applications by generating HTML and other assets at build time, minimizing HTML generation on user devices. We can add dynamic functionality selectively, and deliver smaller bundles of JavaScript. Server-side processing at build time and run time handle business logic and provide just enough data to our frontends only when they need it.
+
+Given that the VA.gov homepage, whose experience is mostly static HTML, currently consists of 4.88 MB of assets (HTML, CSS, JavaScript) (Accessed 24 April 2023) and includes three copies of React in different JavaScript bundles, there is plenty of opportunity for reducing overall page size. If we also [budget for performance](va-rfc-frontend-performance.md), we will have useful goals for application size.
+
+
+## Design
+
+> Explain the proposed design in enough detail so that a team member will fully understand the implementation. Include a diagram (in the `images` dir) as needed to convey your plans. Use active voice, present tense, and decisive language.
+
+The Platform will provide a minimalist option for application teams to create new applications that prioritize generating HTML at build time and do not require React. To start, this can be done with small adjustments to existing tools, e.g. webpack, to allow application manifests to specify non-JSX entry points, exclude React and related libraries, and allow for VFS teams to use/extend existing EJS templates.
+
+VFS teams using the minimalist option will need to adopt a [performance budget](va-rfc-frontend-performance.md), and ensure that complex logic and data manipulation is performed server-side. Minimalist frontends should be able to fetch data that is immediately renderable to the end user. Since `vets-api` already is already responsible for managing and processing dynamic and personalized data, as well as feature toggles, the backend is already well-suited for generating ready-to-display data for frontends.
```

Comment:
No changes, but just a comment that vets-api is often the problem, specifically that all apps need to hit it before loading to check feature toggles

Reply:
probably doesn't need clarification, just calling out. Feel free to resolve

Reply:
This is an excellent point @timwright12 - one concrete step we could take is to note this issue on the feature toggles documentation and suggest that developers consider storing a static value and referencing that if pushing a deploy is fast enough when changes are needed. (Assuming that an app can opt out of using feature toggles? I've never tried)